File: contest/336/leetcode-6317.py
from typing import List 
from collections import OrderedDict
class Solution:
    def beautifulSubarrays(self, nums: List[int]) -> int:
        hashmap = OrderedDict()
        ans, x = 0, 0
        hashmap[0] = 1
        for i in range(len(nums)):
            x = x | nums[i]
            try:
                if hashmap[x] != hashmap.values()[-1]:
                    ans = ans + hashmap[x]
            except KeyError:
                pass
            try:
                hashmap[x] += 1
            except KeyError:
                pass
        return ans

sol = Solution()
print(sol.beautifulSubarrays([4,3,1,2,4]))

File: contest/336/leetcode-6317.cpp
#include <vector>
#include <map>

class Solution{
    public:
        long long beautifulSubarrays(vector<int>& nums){
            map<int, int>pref;
            long long ans=0, x=0;
            pref[0] = 1;
            for(int i=0; i<nums.size(); i++){
                x ^= nums[i];
                if(pref.find(x)!=pref.end()) ans = ans + pref[x];
                pref[x]+=1;
            }
            return ans;
        }
};

// This worked!